src/sys/socket/sockopt.rs

```diff
@@ -1001,6 +1001,133 @@ sockopt_impl!(
     bool
 );
 
+// Documented on `man 7 netlink`
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Add socket to group
+    NetlinkAddMembership,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_ADD_MEMBERSHIP,
+    libc::c_uint
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Drop socket from group
+    NetlinkDropMembership,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_DROP_MEMBERSHIP,
+    libc::c_uint
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Enable packet info control messages
+    NetlinkPacketInfo,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_PKTINFO,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Send broadcast errors
+    NetlinkBroadcastError,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_BROADCAST_ERROR,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Avoid receiving 'ENOBUFS' errors
+    NetlinkNoEnobufs,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_NO_ENOBUFS,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Listen to notifications from all network namespaces
+    NetlinkListenAllNsid,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_LISTEN_ALL_NSID,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Trim the extack if needed
+    NetlinkCapAck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_CAP_ACK,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Always send extack
+    NetlinkExtAck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_EXT_ACK,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Get strict check
+    NetlinkGetStrictCheck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_GET_STRICT_CHK,
+    bool
+);
+
+/// List netlink groups
+/// The returned Vec contains a bit set in chunks of u32.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[derive(Copy, Clone, Debug)]
+pub struct NetlinkListMemberships;
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+impl GetSockOpt for NetlinkListMemberships {
+    type Val = Vec<u32>;
+
+    fn get(&self, fd: RawFd) -> Result<Self::Val> {
+        let mut len = 0;
+        unsafe {
+            // Get membership length
+            let res = libc::getsockopt(
+                fd,
+                libc::SOL_NETLINK,
+                libc::NETLINK_LIST_MEMBERSHIPS,
+                ::std::ptr::null_mut(),
+                &mut len,
+            );
+            Errno::result(res)?;
+            let mut val = vec![0u32; (len as usize) / mem::size_of::<u32>()];
+            if val.capacity() > 0 {
```

Comment:
Why check its capacity here?  You just defined it one line above.

Reply:
`len` can be 0. If you believe `val.length()` is better suited, let me know!

Reply:
I updated to `.len`